ci/build_container/docker_push.sh

```diff
@@ -17,16 +17,29 @@ if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$want_push" == "true" ]
 then
     if [[ $(git diff HEAD^ ci/build_container/) ]]; then
         echo "There are changes in the ci/build_container directory"
-        echo "Updating lyft/envoy-build image"
         cd ci/build_container
-        ./docker_build.sh
-        docker login -u $DOCKERHUB_USERNAME -p $DOCKERHUB_PASSWORD
-        docker push lyft/envoy-build:$TRAVIS_COMMIT
-        docker tag lyft/envoy-build:$TRAVIS_COMMIT lyft/envoy-build:latest
-        docker push lyft/envoy-build:latest
+        docker login -u "$DOCKERHUB_USERNAME" -p "$DOCKERHUB_PASSWORD"
+
+        for distro in ubuntu centos
+        do
+            echo "Updating lyft/envoy-build-${distro} image"
+            LINUX_DISTRO=$distro ./docker_build.sh
+            docker push lyft/envoy-build-${distro}:$TRAVIS_COMMIT
+            docker tag lyft/envoy-build-${distro}:$TRAVIS_COMMIT lyft/envoy-build-${distro}:latest
+            docker push lyft/envoy-build-${distro}:latest
+
+            if [ "$distro" == "ubuntu" ]
+            then
+                echo "Updating lyft/envoy-build image"
+                docker tag lyft/envoy-build-${distro}:$TRAVIS_COMMIT lyft/envoy-build:$TRAVIS_COMMIT
+                docker push lyft/envoy-build:$TRAVIS_COMMIT
+                docker tag lyft/envoy-build:$TRAVIS_COMMIT lyft/envoy-build:latest
```

Comment:
Is there no way to symlink in DockerHub or the like?

Reply:
i didn't know any symlink tricks, but the same layer will be only push once for docker hub.

```bash
[vagrant@localhost build_container]$ docker tag busybox:latest jwfang/jwfr:v1
[vagrant@localhost build_container]$ docker tag jwfang/jwfr:v1 jwfang/jwfr:latest
[vagrant@localhost build_container]$ docker images | fgrep jwfang
jwfang/jwfr                                                      latest              00f017a8c2a6        6 months ago        1.11MB
jwfang/jwfr                                                      v1                  00f017a8c2a6        6 months ago        1.11MB
[vagrant@localhost build_container]$ docker push jwfang/jwfr:v1
The push refers to a repository [docker.io/jwfang/jwfr]
c0de73ac9968: Pushed

v1: digest: sha256:ea4778496324d258547363c32e51d6d0e698dc61952e3b67b78e0fa00af3cd5b size: 527
[vagrant@localhost build_container]$ docker push jwfang/jwfr:latest
The push refers to a repository [docker.io/jwfang/jwfr]
c0de73ac9968: Layer already exists
latest: digest: sha256:ea4778496324d258547363c32e51d6d0e698dc61952e3b67b78e0fa00af3cd5b size: 527
```